src/languages/vbnet.js

```diff
@@ -1,73 +1,179 @@
-/*
-Language: Visual Basic .NET
-Description: Visual Basic .NET (VB.NET) is a multi-paradigm, object-oriented programming language, implemented on the .NET Framework.
-Author: Poren Chiang <ren.chiang@gmail.com>
-Website: https://docs.microsoft.com/en-us/dotnet/visual-basic/getting-started/
-*/
+/**
+ * Language: Visual Basic .NET
+ * Description: Visual Basic .NET (VB.NET) is a multi-paradigm, object-oriented programming language, implemented on the .NET Framework.
+ * Authors: Poren Chiang <ren.chiang@gmail.com>, Jan Pilzer
+ * Website: https://docs.microsoft.com/dotnet/visual-basic/getting-started/
+ */
 
 export default function(hljs) {
+  /**
+   * Character Literal
+   * Either a single character ("a"C) or an escaped double quote (""""C).
+   */
+  const CHARACTER = {
+    className: 'string',
+    begin: /"(""|[^/n])"C\b/
+  };
+
+  const STRING = {
+    className: 'string',
+    begin: /"/,
+    end: /"/,
+    illegal: /\n/,
+    contains: [
+      {
+        // double quote escape
+        begin: /""/
+      }
+    ]
+  };
+
+  /** Date Literals consist of a date, a time, or both separated by whitespace, surrounded by # */
+  const DATE = {
+    className: 'literal',
+    variants: [
+      {
+        // #YYYY-MM-DD# (ISO-Date)
+        begin: /# *\d{4}(-\d{1,2}){2} *#/
+      },
+      {
+        // #M/D/YYYY# (US-Date)
+        begin: /# *(\d{1,2}\/){2}\d{4} *#/
+      },
+      {
+        // #H:mm[:ss]# (24h Time)
+        begin: /# *\d{1,2}(:\d{1,2}){1,2} *#/
+      },
+      {
+        // #h[:mm[:ss]] A# (12h Time)
+        begin: /# *(\d|1[012])(:\d+){0,2} *(AM|PM) *#/
+      },
+      {
+        // ISO-Date and 24h Time
+        begin: /# *\d{4}(-\d{1,2}){2} +\d{1,2}(:\d{1,2}){1,2} *#/
+      },
+      {
+        // ISO-Date and 12h Time
+        begin: /# *\d{4}(-\d{1,2}){2} +(\d|1[012])(:\d+){0,2} *(AM|PM) *#/
+      },
+      {
+        // US-Date and 24h Time
+        begin: /# *(\d{1,2}\/){2}\d{4} +\d{1,2}(:\d{1,2}){1,2} *#/
+      },
+      {
+        // US-Date and 12h Time
+        begin: /# *(\d{1,2}\/){2}\d{4} +(\d|1[012])(:\d+){0,2} *(AM|PM) *#/
+      }
+    ]
+  };
+
+  const NUMBER = {
+    className: 'number',
+    relevance: 0,
+    variants: [
+      {
+        // Float
+        begin: /-?\b\d[\d_]*((\.[\d_]+(E[+-]?[\d_]+)?)|(E[+-]?[\d_]+))[RFD@!#]?/
+      },
+      {
+        // Integer (base 10)
+        begin: /-?\b\d[\d_]*((U?[SIL])|[%&])?/
```

Comment:
```suggestion
        begin: /\b\d[\d_]*((\.[\d_]+(E[+-]?[\d_]+)?)|(E[+-]?[\d_]+))[RFD@!#]?/
      },
      {
        // Integer (base 10)
        begin: /\b\d[\d_]*((U?[SIL])|[%&])?/
```

We can't do this, no way to tell the difference between `-3` and `2 - 3` so we do not include `-` as part of numbers.

Reply:
That makes sense.

I modeled my regex after the `C_NUMBER_RE` which starts with `-?`:
https://github.com/highlightjs/highlight.js/blob/3a2934d7e7af99631527bf55465fa229522735b9/src/lib/modes.js#L8

Reply:
Oh wow, that would be a mistake.  I created an issue for it.